### DISCUSSION TITLE
[Feature]: Export the whole library

### DISCUSSION THREAD

**Discussion by ajhawley** [immich-app/immich#1644] [Feature Request]
### Feature detail

The ability to download all your media files with one click into a compressed archive

currently google takeout zips the media into folders per year like so

Takeout\Google Photos\Photos from 2015
Takeout\Google Photos\Photos from 2016
Takeout\Google Photos\Photos from 2017

it would be nice if immich could do the same or even better and put them into monthly folders too eg Media\November\
and any undated images could be put into an unknown folder. 

---------------------------

extra options could be to download only images or video.
the ability to download all media but split the media types eg
 /photos/2022/july/
/video/2022/july

thanks for taking time to read

### Platform

Web

**Comment by jrasm91** [comment]
Partial support for this is in the works in #956.

**Comment by dkerlee** [comment]
As I was listening to the youtube interview and announcement with Futo, I remembered the only reason I was able to populate Immich with photos in the first place was GOOGLE TAKEOUT. If all my stuff was locked up in Google Photos, I would only have pics from beginning-of-digi-pic-time through about 2023 when I gave up on backing up my cell phone pics to my home server.

To keep this project ... Open, forkable, and NOT lock-in, the Take-Out feature (please rename) needs to be better than TAKE-OUT.

For instance:

1. have it available
2. select GB chunk zip files
3. open source easy read transportable (also selectable?) database metadata
4. option to merge all postgres database held metadata to the actual asset files

**Comment by bo0tzz** [comment]
As a small note here, Immich's files are freely available on the filesystem in UPLOAD_LOCATION. If you enable storage template Immich also sorts them into a nice folder structure. Rather than downloading an export, you can just pull the files from there.

**Reply by Acidiphilium** [reply]
True but with default template, this upload folder is a mess. Maybe not to make immich work like a charm but to use that folder as "takeout" backup solution. Yes using another template would be a solution but from what I read here and on reddit or even on immich website, using another template is kind of risky and if I'm right, there is no easy downgrade to the default template. That makes me a bit afraid to use that option but maybe I'll try it in the future when immich becomes 100% strongly stable.
